vips/foreign.go

```diff
@@ -131,6 +132,20 @@ func IsTypeSupported(imageType ImageType) bool {
 	return supportedImageTypes[imageType]
 }
 
+// DetermineImageTypeFromSource attempts to determine the image type of the given source
+func DetermineImageTypeFromSource(source *Source) ImageType {
+	length := 20
+	data := C.source_sniff(source.vipsSrc, C.ulong(length))
+	sh := &reflect.SliceHeader{
```

Comment:
Hi @negator, I think you need to add `nil` check here, something like this
```go
if data == nil {
		return ImageTypeUnknown
}
```
I got this error when running test `TestNewImageFromReaderFail` (vips/image_test.go:913):
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x40abb4]
```

Reply:
Great find thank you!